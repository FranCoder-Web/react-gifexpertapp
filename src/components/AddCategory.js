import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategories }) => {
	const [inputValue, setInputValue] = useState('')

	const handleInputChange = (e) => {
		setInputValue(e.target.value)
	}
	const handleSubmit = (e) => {
		e.preventDefault()

		if (inputValue.trim().length > 2) {
			setCategories((prevState) => {
				let repeatCategorie = prevState.indexOf(
					inputValue.toUpperCase().trim()
				)
				if (repeatCategorie >= 0) {
					return prevState
				} else {
					return [inputValue.toUpperCase(), ...prevState]
				}
			})
			setInputValue('')
		}
	}
	return (
		<form onSubmit={handleSubmit}>
			<input
				className='search'
				type='text'
				value={inputValue}
				onChange={handleInputChange}
				placeholder='SEARCH GIFS...'
			/>
		</form>
	)
}

AddCategory.propTypes = {
	setCategories: PropTypes.func.isRequired,
}
