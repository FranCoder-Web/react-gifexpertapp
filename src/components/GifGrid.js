import React from 'react'
import PropTypes from 'prop-types'
import { GifGridItem } from './GifGridItem'
import { useFetchGifs } from '../hooks/useFetchGifs'
import Loader from 'react-loader-spinner'

export const GifGrid = ({ category, cleanCategory }) => {
	const { data: images, loading } = useFetchGifs(category)
	const handleCategory = () => {
		cleanCategory((prevstate) => {
			return prevstate.filter((cat) => cat !== category)
		})
	}

	return (
		<>
			<h2 className='category'>
				{category}
				<span onClick={handleCategory} className='x-category'>
					X
				</span>
			</h2>
			{loading && (
				<Loader
					type='ThreeDots'
					color='#1e272e'
					height={120}
					width={120}
				/>
			)}
			<div className='card-grid'>
				{images.map((img) => (
					<GifGridItem key={img.id} category={category} {...img} />
				))}
			</div>
		</>
	)
}

GifGrid.propTypes = {
	category: PropTypes.string.isRequired,
}
