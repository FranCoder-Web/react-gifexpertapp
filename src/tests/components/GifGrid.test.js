import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import { GifGrid } from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs'
jest.mock('../../hooks/useFetchGifs')

describe('Probando el componente <GifGrid />', () => {
	const category = 'Sao'
	useFetchGifs.mockReturnValue({
		data: [],
		loading: true,
	})

	test('debe mostrarse correctamente', () => {
		const wrapper = shallow(<GifGrid category={category} />)

		expect(wrapper).toMatchSnapshot()
	})

	test('debe mostrar items cuando se cargan imagenes con useFetchGifs', () => {
		const gifs = [
			{
				id: 'algo',
				title: 'otra cosa',
				url: 'https://localhost/cualquier/imagen.jpg',
			},
			{
				id: '123',
				title: 'otra cosa mas',
				url: 'https://localhost/cualquier/imagen2.jpg',
			},
		]

		useFetchGifs.mockReturnValue({
			data: gifs,
			loading: false,
		})

		const wrapper = shallow(<GifGrid category={category} />)

		// expect(wrapper).toMatchSnapshot()

		expect(wrapper.find('p').exists()).toBe(false)
		expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
	})
})
