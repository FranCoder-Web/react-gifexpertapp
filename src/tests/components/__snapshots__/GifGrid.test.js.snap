// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente <GifGrid /> debe mostrarse correctamente 1`] = `
<Fragment>
  <h2
    className="category"
  >
    Sao
  </h2>
  <Loader
    className=""
    color="#1e272e"
    height={120}
    style={Object {}}
    timeout={0}
    type="ThreeDots"
    visible={true}
    width={120}
  />
  <div
    className="card-grid"
  />
</Fragment>
`;
